contracts/smart-contract-wallet/SmartAccountFactory.sol

```diff
@@ -4,70 +4,91 @@ pragma solidity 0.8.12;
+    function deployCounterFactualWallet(address _implementation, address _handler, address _owner, uint256 _index) public returns(address proxy){
+        // check optimisation scope in creating salt...
+        bytes32 salt = keccak256(abi.encodePacked(_handler, _owner, address(uint160(_index))));
+
+        bytes memory deploymentData = abi.encodePacked(type(Proxy).creationCode, uint256(uint160(_implementation)));
+
         // solhint-disable-next-line no-inline-assembly
         assembly {
             proxy := create2(0x0, add(0x20, deploymentData), mload(deploymentData), salt)
         }
         require(address(proxy) != address(0), "Create2 call failed");
-        // EOA + Version tracking
-        emit SmartAccountCreated(proxy,_defaultImpl,_owner, VERSION, _index);
-        BaseSmartAccount(proxy).init(_owner, _defaultFallbackHandler);
-        isAccountExist[proxy] = true;
+
+        // marked for deletion
+        // calldata for init method
+        /*if (initializer.length > 0) {
+            // solhint-disable-next-line no-inline-assembly
+            assembly {
+                if eq(call(gas(), proxy, 0, add(initializer, 0x20), mload(initializer), 0, 0), 0) {
+                    revert(0, 0)
+                }
+            }
+        }*/
+
+        // you can pass initializer data but then that it also needs to be part of salt
+        // init method name subject to change
+        BaseSmartAccount(proxy).init(_owner, _handler);
+
+        emit AccountCreation(proxy, _implementation, _handler, _owner, _index);
     }
 
     /**
-     * @notice Deploys wallet using create and points it to _defaultImpl
+     * @notice Deploys wallet using create and points it to _implementation
+     * @param _implementation accountLogic proxy is going point to
+     * @param _handler default fallback handler
      * @param _owner EOA signatory of the wallet
     */ 
-    function deployWallet(address _owner) public returns(address proxy){ 
-        bytes memory deploymentData = abi.encodePacked(type(Proxy).creationCode, uint(uint160(_defaultImpl)));
+    function deployWallet(address _implementation, address _handler, address _owner) public returns(address proxy){ 
+        bytes memory deploymentData = abi.encodePacked(type(Proxy).creationCode, uint256(uint160(_implementation)));
+
         // solhint-disable-next-line no-inline-assembly
         assembly {
             proxy := create(0x0, add(0x20, deploymentData), mload(deploymentData))
         }
-        BaseSmartAccount(proxy).init(_owner, _defaultFallbackHandler);
-        isAccountExist[proxy] = true;
+        require(address(proxy) != address(0), "Create call failed");
+
+        // you can pass initializer data but then that it also needs to be part of salt
+        // init method name subject to change
+        BaseSmartAccount(proxy).init(_owner, _handler);
+
+        emit AccountCreation(proxy, _implementation, _handler, _owner, 0);
```

Comment:
We should emit different event here. 
As indexer indexing these events will get wrong information if we emit same AccountCreation event with index value as 0, even though in that method index is not used at all during creation.

Reply:
we can emit a different one when CREATE is used